include/patchworkpp/patchworkpp.hpp

```diff
@@ -970,21 +970,22 @@ void PatchWorkpp<PointT>::callbackCloud(const sensor_msgs::msg::PointCloud2::Con
     const auto node_graph = this->get_node_graph_interface();
     if (node_graph->count_subscribers(pub_ground_->get_topic_name()) > 0)
     {
-        pub_ground_->publish(cloud2msg(pc_ground, cloud_msg->header.stamp, output_frame));
+        pub_ground_->publish(std::move(cloud2msg(pc_ground, cloud_msg->header.stamp, output_frame)));
     }
     if (node_graph->count_subscribers(pub_non_ground_->get_topic_name()) > 0)
     {
-        pub_non_ground_->publish(cloud2msg(pc_non_ground, cloud_msg->header.stamp, output_frame));
+        pub_non_ground_->publish(std::move(cloud2msg(pc_non_ground, cloud_msg->header.stamp, output_frame)));
     }
 }
 
-template<typename PointT>
-sensor_msgs::msg::PointCloud2 PatchWorkpp<PointT>::cloud2msg(pcl::PointCloud<PointT> cloud, const rclcpp::Time& stamp, std::string frame_id) {
-    sensor_msgs::msg::PointCloud2 cloud_ROS;
-    pcl::toROSMsg(cloud, cloud_ROS);
-    cloud_ROS.header.stamp = stamp;
-    cloud_ROS.header.frame_id = frame_id;
-    return cloud_ROS;
+template<typename PointT> inline
+sensor_msgs::msg::PointCloud2::UniquePtr PatchWorkpp<PointT>::cloud2msg(pcl::PointCloud<PointT> cloud, const rclcpp::Time& stamp, std::string frame_id)
+{
+    sensor_msgs::msg::PointCloud2::UniquePtr cloud_ROS = std::make_unique<sensor_msgs::msg::PointCloud2>();
+    pcl::toROSMsg(cloud, *cloud_ROS);
+    cloud_ROS->header.stamp = stamp;
+    cloud_ROS->header.frame_id = frame_id;
```

Comment:
I think `toROSMsg` set the stamp and header correctly already

Reply:
The header is empty in the created `pcl::PointCloud<PointT> cloud` and that's why it needs to be set here.

Reply:
oh, ok, that's fine then